spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/PositionDeletesRowReader.java

```diff
@@ -90,15 +95,23 @@ protected CloseableIterator<InternalRow> open(PositionDeletesScanTask task) {
         ExpressionUtil.extractByIdInclusive(
             task.residual(), expectedSchema(), caseSensitive(), Ints.toArray(nonConstantFieldIds));
 
-    return newIterable(
+    if (ContentFileUtil.isDV(task.file())) {
+      DVIterable dvIterable = new DVIterable(inputFile, task.file(), task.spec(), expectedSchema());
+      closeableGroup.addCloseable(dvIterable);
+      return dvIterable.iterator();
+    }
+
+    CloseableIterable<InternalRow> iterable =
+        newIterable(
             inputFile,
             task.file().format(),
             task.start(),
             task.length(),
             residualWithoutConstants,
             expectedSchema(),
-            idToConstant)
-        .iterator();
+            idToConstant);
+    closeableGroup.addCloseable(iterable);
```

Comment:
it seems like the existing iterable was never registered to be closed, so I've fixed that here for the existing ones and for DVs

Reply:
actually this isn't true and the closing is being handled in `BaseReader`